src/main.ts

```diff
@@ -1,21 +1,14 @@
 import { buildSLSAProvenancePredicate } from '@actions/attest'
 import * as core from '@actions/core'
 
-const VALID_SERVER_URLS = [
-  'https://github.com',
-  new RegExp('^https://[a-z0-9-]+\\.ghe\\.com$')
-] as const
-
 /**
  * The main function for the action.
  * @returns {Promise<void>} Resolves when the action is complete.
  */
 export async function run(): Promise<void> {
   try {
-    const issuer = getIssuer()
-
     // Calculate subject from inputs and generate provenance
-    const predicate = await buildSLSAProvenancePredicate(issuer)
+    const predicate = await buildSLSAProvenancePredicate()
```

Comment:
The logic to calculate the correct issuer has been moved into the `buildSLSAProvenancePredicate` function.